File: fizzbuzz/python/ChrisD/fizz.py
import unittest
import sys


def fizz(num):
    for i in range(num+1):
        if (i % 3 == 0):
            print("fizz", end = "")
        if (i % 5 == 0):
            print("buzz", end = "")
        if(i % 3 > 0 and i % 5 > 0 and i % 7 > 0 and i % 11 > 0):
            print(i, end = "")
        if (i % 7 == 0):
            print("ping", end = "")
        if (i % 11 == 0):
            print("pong", end = "")
        print()

fizz(int(sys.argv[1]))

File: fizzbuzz/fortran/brynwilkerson/fizz.f90
MODULE fizz
	contains
	function fizzString(i)
		implicit none
		integer :: i
		Character(len = 20) :: fizzString
		character(8) :: i_char
		fizzString = ""
		if (mod(i,3) == 0) then
			 fizzString = trim(fizzString)//"Fizz"
		end if
		if (mod(i,5) == 0) then
			fizzString = trim(fizzString)//"Buzz"
		end if
		if (mod(i,7) == 0) then
			fizzString = trim(fizzString)//"Ping"
		end if
		if (mod(i,11) == 0) then
			fizzString = trim(fizzString)//"Pong"
		end if
		if (mod(i,3) /= 0 .and. mod(i,5) /= 0 .and. mod(i,7) /= 0 .and. mod(i,11) /= 0) then
			write(i_char, '(i8)') i
			fizzString = trim(fizzString)//adjustl(i_char)
		end if
		return
	end function	
END MODULE fizz
